File: modules/spring-core/testdata/java/inspection/profileWrongPlacement/src/com/ProfilePlacementConfiguration.java
package com;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;
import org.springframework.core.annotation.AliasFor;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.METHOD, ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Profile
@interface ProfileA {
}

@Profile
@ProfileA
@Configuration
public class DependsOnPropertyConfiguration {
    
    @Profile
    @ProfileA
    @Bean
    String valid() {return "";}
    
    @Profile
    @ProfileA
    String notABean() {return "";}
    
}

@Profile
@ProfileA
public class NotAComponent { }

File: modules/spring-core/testdata/kotlin/inspection/profileWrongPlacement/src/com/ProfilePlacementConfiguration.kt
package com

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile

@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.CLASS
)
@Retention(
    AnnotationRetention.RUNTIME
)
@Profile
internal annotation class ProfileA

@Profile
@ProfileA
@Configuration
open class DependsOnPropertyConfiguration {
    @Profile
    @ProfileA
    @Bean
    open fun valid(): String {
        return ""
    }

    @Profile
    @ProfileA
    open fun notABean(): String {
        return ""
    }
}

@Profile
@ProfileA
class NotAComponent